packages/block-editor/src/components/global-styles/hooks.js

```diff
@@ -160,11 +157,9 @@ export function useGlobalStyle(
 		: `styles.blocks.${ blockName }${ appendedPath }`;
 
 	const setStyle = ( newValue ) => {
-		setUserConfig( ( currentConfig ) => {
-			// Deep clone `currentConfig` to avoid mutating it later.
-			const newUserConfig = JSON.parse( JSON.stringify( currentConfig ) );
-			set(
-				newUserConfig,
+		setUserConfig( ( currentConfig ) =>
```

Comment:
Simplifying because we don't need to deep clone - `setImmutably` will already do that.